src/clarity-angular/popover/signpost/if-open.directive.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+    Directive,
+    TemplateRef,
+    ViewContainerRef,
+    Input,
+    OnInit, Output, EventEmitter
+} from "@angular/core";
+
+import { IfOpenService } from "./if-open.service";
+
+@Directive({
+    selector: "[clrIfOpen]"
+})
+
+/**********
+ *
+ * @class IfOpenDirective
+ *
+ * @description
+ * A structural directive that controls whether or not the associated TemplateRef is instantiated or not.
+ * It makes use of a Component instance level service: IfOpenService to maintain state between itself and the component
+ * using it in the component template.
+ *
+ */
+export class IfOpenDirective implements OnInit {
+
+    /*********
+     * @property _open
+     *
+     * @description
+     * A property that holds the open or closed state for the template implementing this directive.
+     *
+     * @type {boolean}
+     * @private
+     */
+    private _open: boolean = false;
```

Comment:
The service tracks the state already, you don't want 2 sources of truth. This directive should just use `this.ifOpenService.open` when needed.